packages/components/pagination/src/pagination.tsx

```diff
@@ -239,7 +239,10 @@ export default function Pagination({
         <div>Showing</div>
         <Select
           defaultValue={String(currentItemPerPage)}
-          onChange={(value) => setCurrentItemPerPage(Number(value))}
+          onChange={(value) => {
+            setCurrentPage(1)
+            setCurrentItemPerPage(Number(value))
+          }}
```

Comment:
I think we should expose the props `itemPerPage` and `setItemPerPage` rather than hardcoding the value change inside the component